src/Emulator/Extensions/Mocks/HPSHostController.cs

```diff
@@ -171,6 +174,114 @@ private byte[] GetBytesFromSlave(int count, TimeInterval timeInterval)
             return result;
         }
 
+        private void PollForError(TimeInterval timeInterval){
+            var errors = ReadCommonErrorStatusBytes(timeInterval);
+            var sw = new Stopwatch();
+            sw.Start();
+            while (errors[0] == 0x00 && sw.Elapsed < new TimeSpan(0,0,3)){
+                errors = ReadCommonErrorStatusBytes(timeInterval);
+            }
+        }
+
+        // When we need to read for more than 255 bytes, we will need this method
+        private byte[] GetLongBytes(int count, TimeInterval timeInterval) {            
+            var result = new byte[count];
+            int rounds = count /255;
+            int remainder = count % 255; 
+            for (int i = 0; i <= rounds; i ++) {
+                int toRead = i == rounds ? remainder : 255;
+                Array.Copy(GetBytesFromSlave(toRead, timeInterval), 0, result, i * 255, toRead);
+            }
+            return result;
+        }
+
+        public void StartSocketServer(TimeInterval timeInterval)
+        {  
+            Console.WriteLine(currentSlave.GetType().Name);
+            string path = "/tmp/i2c.sock";
+            if (File.Exists(path))
+            {
+                File.Delete(path);
+            }
+            var socket = new Socket(AddressFamily.Unix, SocketType.Stream, ProtocolType.Unspecified);
+            socket.Bind(new UnixEndPoint(path));
```

Comment:
Due to the use of Unix sockets, this whole code will have to be compiled conditionally not to fail on Windows.

I guess a relatively simple change would be to change it into an `IPEndPoint`. Would that work for you?

It would boil down to

```
  var port = SomeNumber;
  var socket = new Socket(AddressFamily.InterNetwork, SocketType.Stream, ProtocolType.Tcp);
  socket.Bind(new IPEndPoint(IPAddress.Any, port));
```

Of course I understand that the use case here is a bit specific and depending on the application on the other side of the socket, so whichever option (ifdef or switching to a port) suits you better is fine.

Reply:
Used ifdef, as unixsocket has been implemented on the other side, and it's lighter and faster in this scenario.